File: DSA Practice/LeetCode/509. Fibonacci Number/solution.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0:
            return 0
        
        elif n == 1:
            return 1

        else :
            a = 0
            b = 1

            for _ in range(2, n+1):
                temp = a
                a = b
                b += temp

            return b

File: DSA Practice/LeetCode/509. Fibonacci Number/solution.java
class Solution {
    public int fib(int n) {
        if (n == 0){
            return 0;
        } else if (n == 1){
            return 1;
        } else {
            int a = 0;
            int b = 1;

            for (int i = 2; i < n+1; i++){
                int temp = a;
                a = b;
                b = b + temp;
            }

            return b;
        }
    }
}